Bluetooth: BTP: Extend BTP with Mesh related commands

MariuszSkamra opened the pull request:
This adds BTP commands needed to test Mesh against PTS

MariuszSkamra committed:
Bluetooth: btp: Add BTP Mesh Network Send command

This command will be used to verify functionality of the network
layer against PTS.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra committed:
Bluetooth: BTP: Add BTP MESH IV Update related commands

This commands will be used while testing IV Update procedure
against PTS.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

jhedberg commented on tests/bluetooth/tester/btp_spec.txt:
```diff
@@ -1391,6 +1391,44 @@ Commands and responses:
+		Command parameters:	Valid parameters (1 octet)
+					TTL (1 octet)
+					SRC (2 octets)
+					DST (2 octets)
```
Isn't there the actual payload parameter missing here?

MariuszSkamra replied:
I'll explain myself ;) I thought about adding this, but I saw that PTS is not for any particular data to be sent, so I just skipped this and I wanted to add payload in handler of this command.
But on the other hand this command would be more versatile when payload will be here.

jhedberg commented on tests/bluetooth/tester/btp_spec.txt:
```diff
@@ -1391,6 +1391,44 @@ Commands and responses:
+
+	Opcode 0x0a - Network Send
+		Controller Index:	<controller id>
+		Command parameters:	Valid parameters (1 octet)
```
What does "Valid parameters (1 octet)" mean?

MariuszSkamra replied:
TTL parameter is not always given in PTS MMI, so I needed to some bitfield to indicate which paramaters has to be taken into account

MariuszSkamra commented:
updated

MariuszSkamra committed:
Bluetooth: btp: Add Mesh Network receive event

This event is needed for qualification purposes to verify functionality
of Mesh Network layer.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra commented:
few mode commands & events added

MariuszSkamra committed:
Bluetooth: btp: Introduce Health Generate/Clear Faults commands

Those commands will be used for qualification purposes to verify
the functionality of Mesh Health Model.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra commented:
Health Generate Faults & Health Clear Faults commands added

MariuszSkamra commented:
rebased

MariuszSkamra commented:
rebased

jhedberg commented on tests/bluetooth/tester/btp_spec.txt:
```diff
@@ -1510,3 +1569,15 @@ Events:
+		Event parameters:	TTL (1 octet)
+					CTL (1 octet)
+					SRC (2 octets)
+					DST (2 octets)
```
The chosen information seems a bit arbitrary to me. So we don't need information on which IVI was used, which NetKey Index was used, etc? I suppose the payload is the payload after decryption with the network key? Don't some tests also require verification of data after app key decryption?

MariuszSkamra replied:
Of course, we can extend the event information if it'll be needed. I choose those, to satisfy PTS needs.
````
Please confirm the following network packet was received: Network Header: 
    TTL: [0x00]
    CTL: [0x00]
    SRC: [0x0001]
    DST: [0xFFFC]
    TransportPDU: [0x01020304050607080910111213141516]
````
Yes, the payload here is the payload is after decryption.
I did not saw such tests yet.

jhedberg replied:
After decryption with the NetKey or after decryption with NetKey+AppKey?

MariuszSkamra replied:
decryption with the netkey
